src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -450,11 +462,11 @@ class GroupReadsByUmi
     logger.info("Filtering and sorting input.")
     in.iterator
       .filter(r => !r.secondary && !r.supplementary)
-      .filter(r => (includeNonPfReads || r.pf)            || { filteredNonPf += 1; false })
-      .filter(r => (r.paired && r.mapped && r.mateMapped) || { filteredPoorAlignment += 1; false })
-      .filter(r => (r.refIndex == r.mateRefIndex)         || { filteredPoorAlignment += 1; false })
-      .filter(r => (r.mapq >= this.minMapQ && r.get[Int](SAMTag.MQ.name()).exists(_ >= this.minMapQ)) || { filteredPoorAlignment += 1; false })
-      .filter(r => !r.get[String](rawTag).exists(_.contains('N')) || { filteredNsInUmi += 1; false })
+      .filter(r => (includeNonPfReads || r.pf)                      || { filteredNonPf += 1; false })
+      .filter(r => (r.mapped && (!r.paired || r.mateMapped))        || { filteredPoorAlignment += 1; false })
+      .filter(r => (!r.paired || r.refIndex == r.mateRefIndex)      || { filteredPoorAlignment += 1; false })
+      .filter(r => mapqOk(r, this.minMapQ)                          || { filteredPoorAlignment += 1; false })
```

Comment:
Can anything that increments `filteredPoorAlignment` just be refactored into one method?

Reply:
I just spent 5 minutes trying to do this and I feel like the result is worse.  It's actually helpful to apply the filters stepwise so you don't call things that don't make sense on certain reads.